### PR TITLE
Add 12 more grammar errors

### PR DESCRIPTION
**Pull request by sandersn**
These are the last ones that I know of. They come from calls to `grammarErrorOnFirstToken`.

Fixes part of #45349
Follow-up to #47067

### CODE REVIEW

File: src/compiler/program.ts [hunk 1]
```diff
@@ -855,13 +857,21 @@ namespace ts {
         Diagnostics.A_rest_parameter_cannot_have_an_initializer.code,
         Diagnostics.A_rest_parameter_must_be_last_in_a_parameter_list.code,
         Diagnostics.A_rest_parameter_or_binding_pattern_may_not_have_a_trailing_comma.code,
+        Diagnostics.A_return_statement_can_only_be_used_within_a_function_body.code,
+        Diagnostics.A_return_statement_cannot_be_used_inside_a_class_static_block.code,
         Diagnostics.A_set_accessor_cannot_have_rest_parameter.code,
         Diagnostics.A_set_accessor_must_have_exactly_one_parameter.code,
+        Diagnostics.An_export_declaration_can_only_be_used_in_a_module.code,
+        Diagnostics.An_export_declaration_cannot_have_modifiers.code,
+        Diagnostics.An_import_declaration_can_only_be_used_in_a_namespace_or_module.code,
```
Comment (andrewbranch) [inline src/compiler/program.ts:866]: This wording seems off, and we could probably improve it and make it friendlier to JS (which doesn’t have namespaces) at the same time. I think it means that an import declaration has to be at the top level of the file.

Comment (sandersn) [reply src/compiler/program.ts:866]: That's how it's implemented, yes. Well, it also checks for parents like `namespace N` or `declare module "m"`, neither of which apply to JS.
I'll open a bug proposing to split the current implementation into 2 messages and improve the wording.

**Review by andrewbranch — APPROVED** [review]

File: src/compiler/program.ts (hunk 1 above)
Comment (sandersn) [reply src/compiler/program.ts:866]: #47076

File: src/compiler/program.ts [hunk 2]
```diff
@@ -855,13 +857,21 @@ namespace ts {
...
         Diagnostics.A_set_accessor_must_have_exactly_one_parameter.code,
+        Diagnostics.An_export_declaration_can_only_be_used_in_a_module.code,
+        Diagnostics.An_export_declaration_cannot_have_modifiers.code,
+        Diagnostics.An_import_declaration_can_only_be_used_in_a_namespace_or_module.code,
+        Diagnostics.An_import_declaration_cannot_have_modifiers.code,
         Diagnostics.An_object_member_cannot_be_declared_optional.code,
         Diagnostics.Argument_of_dynamic_import_cannot_be_spread_element.code,
         Diagnostics.Cannot_assign_to_private_method_0_Private_methods_are_not_writable.code,
         Diagnostics.Cannot_redeclare_identifier_0_in_catch_clause.code,
+        Diagnostics.Catch_clause_variable_cannot_have_an_initializer.code,
```
Comment (amcasey) [inline src/compiler/program.ts:872]: 😆

**Pull request merged**